cal/cal/models.py

```diff
@@ -605,7 +605,11 @@ def get_events(self, calendar_ids=None, start=None, end=None):
 
     def query(self, calendar_ids=None, start=None, end=None):
         if self.calendar:  # Calendar Category
-            calendars = [self.calendar]
+            calendar_in_ids = calendar_ids is not None and self.calendar.calendar_id in calendar_ids
+            if calendar_ids is None or calendar_in_ids:
```

Comment:
if not calendar_ids or (calendar_ids and self.calendar.calendar_id in calendar_ids)